### PR TITLE
cleaned up movement decision in Bot class

### PR DESCRIPTION
**Pull request by hippi345**
reduced variables declared and cleaned up the logic and formatting so its easier to follow
things to think about: can we make the decision for preferred movements be done such that its not dependent on the for loop over the movements made?

### CODE REVIEW

File: src/sample/models/Bot.java [hunk 1]
```diff
@@ -16,24 +16,23 @@ public Bot(int movementBoundary)
...
 
     // random bot movement
     public void MoveRandomly()
     {
         // Determine which moves are possible, then add them to the list
-        ArrayList<MoveEnum> possibleMoves = getPossibleMoves();
-        getPreferredMoves();
+        possibleMoves = getPossibleMoves();
+        preferredMoves = getPreferredMoves();
         Random randomNumberGenerator = new Random(System.currentTimeMillis());
```
Comment (AndreyLebedenko) [inline src/sample/models/Bot.java]: Better keep randomNumberGenerator  as a private field populated in the constructor (or even as a static).

Comment (hippi345) [reply src/sample/models/Bot.java]: good point. fixed that and committed.

File: src/sample/models/Bot.java [hunk 2]
```diff
@@ -33,10 +34,10 @@ public void MoveRandomly()
         // Determine which moves are possible, then add them to the list
         possibleMoves = getPossibleMoves();
         preferredMoves = getPreferredMoves();
-        Random randomNumberGenerator = new Random(System.currentTimeMillis());
+        // Random randomNumberGenerator = new Random(System.currentTimeMillis());
```
Comment (RiskRunner0) [inline src/sample/models/Bot.java]: You can just delete it

File: src/sample/models/Bot.java [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 
-import java.lang.reflect.Array;
 import java.util.ArrayList;
 import java.util.Random;
 
 public class Bot extends Point
 {
     private ArrayList<MoveEnum> preferredMoves = new ArrayList<>();
     private ArrayList<String> movesMade = new ArrayList<>();
+    private ArrayList<MoveEnum> possibleMoves = new ArrayList<>();
```
Comment (RiskRunner0) [inline src/sample/models/Bot.java]: Possible moves needs to be calculated each time, it makes more sense to scope it to a movement method than to have it be a member variable, and clear it every time.

**Review by RiskRunner0 — APPROVED** [review]

File: src/sample/models/Bot.java (hunk 3 above)
Comment (hippi345) [reply src/sample/models/Bot.java]: Addressed by making variables locally scoped

**Review by hippi345 — COMMENTED** [review]
Reviewed changes and recommendations from contributors

**Pull request merged**